### ISSUE TITLE
`pyproject.toml` is ignored when passed via stdin

### ISSUE THREAD

**Issue by InSyncWithFoo** [astral-sh/ruff#16950]
This works:

```shell
$ cat pyproject.toml
[project]
name = 1

$ ruff check --isolated --select RUF200 pyproject.toml
pyproject.toml:2:8: RUF200 Failed to parse pyproject.toml: invalid type: integer `1`, expected a string
  |
1 | [project]
2 | name = 1
  |        ^ RUF200
  |

Found 1 error.
```

This doesn't:

```shell
$ ruff check --isolated --select RUF200 --stdin-filename pyproject.toml -
[project]
name = 1
All checks passed!
```

These two commands should work similarly.

**Comment by MichaReiser** [comment]
> I think the server also doesn't report `RUF200` when a `pyproject.toml` is opened either.

Yes, the server doesn't run on `toml` files
